dvc/external_repo.py

```diff
@@ -62,16 +62,33 @@ def _external_repo(url=None, rev=None, cache_dir=None):
 
     # Adjust new clone/copy to fit rev and cache_dir
     repo = Repo(new_path)
+    original_repo = None
```

Comment:
No need to spread the context of that so widely, better move original_repo closing logic under the `if os.path.isdir()`